extensions/sql-database-projects/src/models/project.ts

```diff
@@ -67,14 +74,24 @@ export class Project {
 	}
 
 	/**
-	 * Writes a file to disk, adds that file to the project, and writes it to disk
+	 * Writes a file to disk if contents are provided, adds that file to the project, and writes it to disk
 	 * @param relativeFilePath Relative path of the file
 	 * @param contents Contents to be written to the new file
 	 */
-	public async addScriptItem(relativeFilePath: string, contents: string): Promise<ProjectEntry> {
+	public async addScriptItem(relativeFilePath: string, contents?: string): Promise<ProjectEntry> {
 		const absoluteFilePath = path.join(this.projectFolderPath, relativeFilePath);
-		await fs.mkdir(path.dirname(absoluteFilePath), { recursive: true });
-		await fs.writeFile(absoluteFilePath, contents);
+
+		if (contents) {
+			await fs.mkdir(path.dirname(absoluteFilePath), { recursive: true });
+			await fs.writeFile(absoluteFilePath, contents);
+		}
+
+		//Check that file actually exists
+		let exists = utils.exists(absoluteFilePath);
+		if (!exists) {
+			//TODO: Should we exit from here?
+			this.apiWrapper.showErrorMessage(constants.noFileExist(absoluteFilePath));
```

Comment:
I think exiting via throwing an Error if the file doesn't exist is good.  That way this class can stay a model and we can keep references to apiWrapper in the more-coupled controller classes